### ISSUE TITLE
[New Device Support] Amazfit Equator && Amazfit Moonbeam

### ISSUE THREAD

**Issue by mad-gooze** [Freeyourgadget/Gadgetbridge#940]
https://xiaomi-mi.com/amazfit/xiaomi-amazfit-equator-black/
https://xiaomi-mi.com/amazfit/xiaomi-amazfit-moon-beam-white/

Amazfit app for this devices does not support app notifications. Gadgetbridge could greatly help here.

I own Amazfit Equator and I tried to examine it with BLE Scanner - it is possible to connect, make it vibrate and read some values.
I can help to add the support for these devices or maybe try to do it myself if someone helps me with a good tutorial.

**Comment by ashimokawa** [comment]
I think these are Mi Bands in another shape. Can you just try to press the + button in Gadgetbridge and see if it shows up and can be paired?

**Comment by ashimokawa** [comment]
@mad-gooze 
Yep, 99% sure this is a Mi Band 1 technically, fw looks almost the same.

**Comment by mad-gooze** [comment]
No, that does not work - it does not get found

**Comment by ashimokawa** [comment]
Please also post a screenshot of the device info (pressing the three dots when GB is connected to the device)

**Comment by mad-gooze** [comment]
Sorry, I as wrong
I can see the device in search, but pairing hangs up (amazfit does not vibrate and nothing happens)
<details>
<img src="https://user-images.githubusercontent.com/1188779/34919828-023ff3f4-f97a-11e7-9fb4-5474b267a08b.jpg"/>
<img src="https://user-images.githubusercontent.com/1188779/34919830-064fc280-f97a-11e7-80d5-1e357fffe2f0.jpg"/>
</details>

**Comment by ashimokawa** [comment]
Can you send the gadgetbridge logs?

**Comment by mad-gooze** [comment]
[gadgetbridge.log](https://github.com/Freeyourgadget/Gadgetbridge/files/1629811/gadgetbridge.log)

**Comment by ashimokawa** [comment]
In mibandsupport.java can you remove the line


  .setWearLocation(builder)


and recompile?

**Comment by ashimokawa** [comment]
This is actually a duplicate of #274

**Comment by mad-gooze** [comment]
Nothing changed after removing `.setWearLocation(builder)`, here are new logs.
[gadgetbridge-no-setWearLocation.log](https://github.com/Freeyourgadget/Gadgetbridge/files/1629926/gadgetbridge.1.log)

**Comment by ashimokawa** [comment]
Can you remove the if block here:

https://github.com/Freeyourgadget/Gadgetbridge/blob/23c6219cefd54b3322bef63b644c7c2a21272234/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/devices/miband/UserInfo.java#L105


and recompile?

The device *IS* very similar to Mi Band, but there is some small detail we obviously have to do differently.

**Comment by cpfeiffer** [comment]
None of the developers have had access to an Amazfit necklace, not back then, nor the new one now. But since you seem to be able to compile Gadgetbridge and modify some source code, we might be able to get it to work!

You could start by commenting out some further invocations in MiBandSupport.initializeDevice() and see how far you get (check logs for where it breaks).

**Comment by mad-gooze** [comment]
Do you have something like Slack to discuss it in a more comfortable way? I can get back to the experiments in weekends

**Comment by PsymonDK** [comment]
I've just gotten an Equator and would love to help with this. Sadly enough I don't have any experience with coding android apps so I don't even know how to start. 
mad-gooze did you ever make any progress?

**Comment by mad-gooze** [comment]
No, I gave it up:(

**Comment by KatElizabeth** [comment]
I'd love to see this working too, as I just got the equator and their Amazfit app is completely worthless (crashing every time it tries to pair, etc etc). I'd be willing to help with this in any way that I can. I do have Android Studio and am pretty familiar with java, but don't have experience testing anything like this.
